src/transport/raw/TCP.cpp

```diff
@@ -343,6 +343,11 @@ CHIP_ERROR TCPBase::ProcessReceivedBuffer(Inet::TCPEndPoint * endPoint, const Pe
             // We have not yet received the complete message.
             return CHIP_NO_ERROR;
         }
+        if (messageSize == 0)
+        {
+            // No payload but considered a valid message. Return success to keep the connection alive.
+            return CHIP_NO_ERROR;
```

Comment:
But ... doesn't this leave the four bytes `0, 0, 0, 0` in state->mReceived?  As in, shouldn't this check-and-return be _after_ the Consume call?

Reply:
That is right. Good catch! Since this is a complete packet, the size field should be consumed first(contrary to the previous if-check-and-return).

Reply:
Thanks. Fixed it.